src/components/patterns/on-the-page.tsx

```diff
@@ -0,0 +1,113 @@
+"use client"
+
+import React, {useState, useEffect, useMemo} from "react"
+import parse from "html-react-parser"
+import {NodeStanfordPage} from "@/lib/gql/__generated__/drupal.d"
+import {ParagraphStanfordWysiwyg} from "@/lib/gql/__generated__/drupal.d"
+import {twMerge} from "tailwind-merge"
+import DrupalLink from "./elements/drupal-link"
+
+interface Heading {
+  text: string
+  id: string
+}
+
+const OnThePageLink = ({node}: {node: NodeStanfordPage}) => {
+  const [activeHeading, setActiveHeading] = useState<string>("")
+  const relLinkHeading = node.sulRelLinksHeading
+  const relLinks = node.sulRelLinks
+
+  const headings = useMemo(() => {
```

Comment:
instead of pulling from the node, let's just look at the page itself using some plain javascript. This will avoid the complexity with behaviors and wysiwyg content.

Reply:
You can use something simple like `document.querySelector('#main-content h2')`

Reply:
Oh that would be perfect especially since I was encountering issues with conditionally determining whether the block was on a Stanford Basic Page or Stanford News Page (e.g. suPageComponents vs suNewsComponents). 

I'll rework using good ole vanilla javascript and have this PR updated in a bit!

Thank you for the speed feedback and nudge in the right direction, @pookmish 😄